backend/src/Designer/Controllers/PreviewController.cs

```diff
@@ -905,5 +954,33 @@ private string ReplaceIndexToFetchCorrectOrgAppInCshtml(string originalContent)
 
             return modifiedContent;
         }
+
+        /// <summary>
+        /// Method to get the mocked altinn nuget build from the version
+        /// We are returnning the minimum BUILD version of the nuget package that is required for app frontend to work
+        /// from v4 and above.
+        /// </summary>
+        /// <param name="version">The version of the nuget package</param>
+        /// <returns>The minimum build version of the nuget package</returns>
+        private string GetMockedAltinnNugetBuildFromVersion(string version)
+        {
+
+            string[] versionParts = version.Split('.');
+            if (versionParts.Length < 3 || Convert.ToInt32(versionParts[0]) < 8)
+            {
+                return string.Empty;
+            }
+
+            if (versionParts[2].Contains("-preview") && versionParts.Length == 4)
+            {
+                int previewVersion = Convert.ToInt32(versionParts[3]);
+                if (previewVersion < MINIMUM_PREVIEW_NUGET_VERSION)
+                {
+                    return string.Empty;
+                }
+            }
+
+            return MINIMUM_NUGET_VERSION;
+        }
```

Comment:
I hope I have understood the code correctly, but would it be more self-explanatory if we rewrote it to something like this? 😊

```suggestion
private string GetMockedAltinnNugetBuildFromVersion(string version)
{
    string[] versionParts = version.Split('.');

    if (!IsVersionValidSemVer(versionParts))
    {
        return string.Empty;
    }

    if (IsPreviewVersion(versionParts))
    {
        int previewVersion = GetPreviewVersion(versionParts);
        if (previewVersion < MINIMUM_PREVIEW_NUGET_VERSION)
        {
            return string.Empty;
        }
    }

    return MINIMUM_NUGET_VERSION;
}

private bool IsVersionValidSemVer(string[] versionParts)
{
    return versionParts.Length >= 3 && Convert.ToInt32(versionParts[0]) >= 8;
}

private bool IsPreviewVersion(string[] versionParts)
{
    return versionParts.Length == 4 && versionParts[2].Contains("-preview");
}

private int GetPreviewVersion(string[] versionParts)
{
    return Convert.ToInt32(versionParts[3]);
}

```